src/gf_ring.cpp

```diff
@@ -59,25 +59,25 @@ void RingModN<uint32_t>::sub_two_bufs(
     uint32_t* res,
     size_t len)
 {
-    simd::u32::sub_two_bufs(bufa, bufb, res, len, this->_card);
+    simd::sub_two_bufs_u32(bufa, bufb, res, len, this->_card);
 }
 
 template <>
 void RingModN<uint16_t>::add(int n, uint16_t* x_u16, uint16_t* y_u16)
 {
     int half = n / 2;
     uint16_t* x_next = x_u16 + half;
-    simd::u16::add_two_bufs(y_u16, x_u16, half, this->_card);
-    simd::u16::add_two_bufs(y_u16, x_next, half, this->_card);
+    simd::add_two_bufs_u16(y_u16, x_u16, half, this->_card);
+    simd::add_two_bufs_u16(y_u16, x_next, half, this->_card);
 }
 
 template <>
 void RingModN<uint32_t>::add(int n, uint32_t* x_u32, uint32_t* y_u32)
```

Comment:
The code for `add` and `hadamard_mul` is the same (after you remove the suffix) for both u16 and u32, I think it can be factorized by using `std::enable_if` + defining a trait such as `is_simd_accelerated`.

This can be done in another PR (when you'll add support for u64, to avoid having code triplicated).

Reply:
okay, I see. It will be optimized in other PR